api/src/main/java/org/apache/iceberg/actions/RepairManifests.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.ManifestFile;
+
+/**
+ * An action that will repair manifests. Implementations should produce a new set of manifests where
+ * duplicate entries are removed and entries which refer to files which no longer exist are removed.
```

Comment:
The interface is prescriptive on what implementations should do in terms of duplicate entries and what implementations should do when files which no longer exist. After some thought I think it's right to define that but I can see arguments where someone wants implementations to just fail in case there are entries which refer to files which no longer exist (Note: I'm specifically talking about cases where implementations cannot recover those files).

Also I think at some level the interface needs to be prescriptive to have a clear defined `Result`.

Reply:
So I think we may be missing a few other things here and I'm not sure if they should all happen on every invocation. In my mind

1. Read metadata from data files and recompute values in manifests
2. Remove Manifests or files with FNF
3. Removing duplicates

I think we probably also want to be able to do a dry run of this? Something that checks these 3 things but doesn't actually commit any changes?

Reply:
NVM (dry run is in the api below)

Reply:
Thanks @amogh-jahagirdar for this.  Yea  Russell's list sounds good, optionally, I guess we can put the options in later when we implement them.

Reply:
@RussellSpitzer @szehon-ho  I largely wanted to avoid a user having to think through which options they need to specify, and make sure implementations have sane defaults but I think I agree at least at the actions API level we could have separate configuraiton methods. When the procedure implementation is there, users could pass in what they want and the procedure implementation has some sane defaults (power users who use the raw actions APIs would have to set every option they want explicitly). cc @danielcweeks in case he had any thoughts.